pkg/converters/gateway/gateway.go

```diff
@@ -451,12 +453,32 @@ func (c *converter) applyCertRef(source *Source, listener *gatewayv1alpha2.Liste
 			source, listener.Name)
 		return
 	}
-	// TODO Support more certificates
-	if len(certRefs) > 1 {
-		err := fmt.Errorf("listener currently supports only the first referenced certificate")
-		c.logger.Warn("skipping one or more certificate references on %s listener '%s': %s",
-			source, listener.Name, err)
+	// iterate over all certRefs
+	for _, certRef := range certRefs {
+		crtFile, err := c.readCertRef(source.namespace, certRef)
+		if err != nil {
+			c.logger.Warn("skipping certificate reference on %s listener '%s': %s",
+				source, listener.Name, err)
+			return
+		}
+		// in crtFile.CommonName replace wildcards * with common regex wildcard .*
+		crtFilePattern := strings.ReplaceAll(crtFile.CommonName, "*", ".*")
```

Comment:
I'd use `HasPrefix("*.")` and, if true, use the resulting of a `TrimPrefix("*.")` into a `HasSuffix()` instead of parse and match a regex on every iteration. This is the hot path of a cluster with dozens of thousands of httproutes, so we want all the processing to be as fast as possible.